### PR TITLE
Update Mitiq paper code blocks based on paper review

### PR DESCRIPTION
**Pull request by andreamari**
Description
-----------

This PR updates the "Mitiq paper code blocks" example, following the suggestions of a paper report.

License
-------

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

### CODE REVIEW

File: docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md [hunk 1]
```diff
@@ -236,9 +237,36 @@ scaled = zne.scaling.scale_parameters(circuit2, scale_factor=2.0, base_variance=
...
+
+```{code-cell} ipython3
+from mitiq.zne.scaling import compute_parameter_variance
+
+# Estimate base level of parameter noise
+base_variance = compute_parameter_variance(executor, cirq.X, cirq.LineQubit(0))
+print("Estimation of parameter noise variance:", base_variance)
+
+def scale_param_noise(circuit, scale_factor):
+    return zne.scaling.scale_parameters(circuit, scale_factor, base_variance=0.009)
```
Comment (rmlarose) [inline docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md:254]: Could use a `lambda` or `functools.partial`.

File: docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md [hunk 2]
```diff
@@ -236,9 +237,36 @@ scaled = zne.scaling.scale_parameters(circuit2, scale_factor=2.0, base_variance=
...
+)
+# We are applying parameter-noise mitigation with an executor which is instead subject to depolarizing noise.
+# Therefore, we don't expect good performances.
+print("ZNE value via parameter noise scaling:", zne_value)
+```
+
 +++ {"id": "ye9I9WLNrGxU"}
 
-**Codeblock 13**
+**Codeblock 14**
```
Comment (rmlarose) [inline docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md:269]: 1. Nice catch.
2. Sucks this has to be done manually.

File: docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md (hunk 1 above)
Comment (andreamari) [reply docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md]: done [3cb771c](https://github.com/unitaryfund/mitiq/pull/1357/commits/3cb771c5fd9b271efa309a3b291a95a18f18fe7c)

File: docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md (hunk 2 above)
Comment (andreamari) [reply docs/source/examples/mitiq-paper/mitiq-paper-codeblocks.md:268]: Hopefully, after the paper is published, they won't be changed for all eternity :-D

**Pull request merged**